### PR TITLE
Implement Password Reset Feature.

### PR DESCRIPTION
**Pull request by Tiger-Cross**

### CODE REVIEW

File: server/repositories/ForgotPasswordTokenRepository.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const {ForgotPasswordToken} = require("../models");
+const Q = require("q");
+const ForgotPasswordTokenRepositoryInterface = require("./interfaces/ForgotPasswordTokenRepositoryInterface");
+
+let ForgotPasswordTokenRepository = Object.create(
+  ForgotPasswordTokenRepositoryInterface
+);
+
+InvitationTokenRepository.add = function(email, token, expires) {
```
Comment (wburr10) [inline server/repositories/ForgotPasswordTokenRepository.js]: Should be ForgotPasswordTokenRepository

Comment (Tiger-Cross) [reply server/repositories/ForgotPasswordTokenRepository.js]: Fixed

File: server/controllers/authController.js [hunk 2]
```diff
@@ -173,13 +175,40 @@ This link will expire in 24 hours.`)
...
+      userRepository.getByEmail(req.body.email)
+        .then(user => {
+          if (user) {
+            const token = crypto.randomBytes(16).toString('hex');
+            const expires = moment().add(30, 'minutes').format();
+            return forgotPasswordTokenRepository.add(req.body.email, token, expires)
+              .then(() => {
+                res.status(200).json({message: "Instructions to reset your password have been sent to the email entered if an account with that email exists."});
+                const emailClient = new EmailClient(emailClientTypes.NOREPLY);
+                return emailClient.send(req.body.email,
```
[the remark this answers is not in the document]
Comment (Tiger-Cross) [reply server/controllers/authController.js]: Fixed thanks to you :)

File: server/controllers/authController.js [hunk 3]
```diff
@@ -173,13 +175,40 @@ This link will expire in 24 hours.`)
...
+Please click the following link to create a new password. 
+Please try and use a password different to any that you have used previously.
+
+${process.env.WEB_URL}/password-reset?token=${token}
+
+This link will expire in 30 minutes.`)
+              });
+          }
+          res.status(200).json({message: "Instructions to reset your password have been sent to the email entered if an account with that email exists."});
+        })
```
[the remark this answers is not in the document]
Comment (Tiger-Cross) [reply server/controllers/authController.js]: Yeah my bad, cheers for reminding

**Pull request merged**